### DISCUSSION TITLE
White Screen with WT32 SC01

### DISCUSSION THREAD

**Discussion by fgathercoupa** [HASwitchPlate/openHASP#267] [Hardware]
Hi, 
first, let me thank you for this great library. 

However, I flashed it on a WT32-SC01 of mine, which works with other binaries quite fine and I find a strange behaviour. On the first start, everything works as expected. However, after the first reset (SW or HW button) I only see a white screen. When I cut the power, wait for 5s, everything works during that boot. Has anyone else seen this? Maybe some timing problem during the display initialization? I already tried all USB cables and power supplies, but so far no luck.

**Comment by DanBendavid** [comment]
My WT32-SC01 with 16Mo is working fine

**Comment by htvekov** [comment]
I've three wt32-sc01 plates here. Has been working solid here for some 10 months now.
Always on 24/7 and has been reflashed sh*tloads of times. Never had any hw issues with these devices

**Reply by filmchaser1** [reply]
Sorry to vent but can you or someone, anyone, please tell me how to get this Wt32-sco01 connected to Home Assistant?! I'm so F---ing mad at the runaround on info. The openHASP page has NO instructions but links to this forum where again there is NO straightforward info on how to connect, the links I find for connecting openHASP to HA say go to HAC (yet another runaround) and lookup openHASP in integrations? But nothing is there, nothing is found in Home Assistant under integrations for openHASP, discovery doesn't see it?WTF...I've wasted hours trying to find valid information....I have HA 2022.2.2 installed on a pi and openHASP installed on the wt32, it displays "plate" onscreen and is connected to my wireless network.

UPDATE: 2 hours later. The instructions are outdated. Nowhere on the openHASP page does it say you must setup MQTT, not some integration from openHASP in either HA or HACs. Once I found that out I installed Mosquito MQTT and had it running in 15 minutes.

**Reply by htvekov** [reply]
Hi'

Sorry to hear that you've spend hours searching for the right information.

openHASP has evolved so fast last year, that it can be quite hard to keep track on all the new stuff appearing.
I guess dev-team struggles even harder to keep documentation up to speed as well.

But openHASP was there long before @dgomes made the nice custom component for HA. And the sw is built with jsonl as the way to define objects/settings and mqtt as the communication platform.

It's actually in the the documentation, but perhaps it should be emphasised a bit more that custom component install is not enough to get openHASP on the road.

Rgds.
Henning

![image](https://user-images.githubusercontent.com/45469813/152675025-5de6d68f-25f4-4e21-8035-b4ee8f861e11.png)

**Reply by nagyrobi** [reply]
@filmchaser1 I'm sorry but it looks like you didn't read carefully enough. It's plain to see:
https://haswitchplate.github.io/openHASP-docs/0.6.2/integrations/home-assistant/howto/#configuration

Quote from that page:

> Configuration[~](https://haswitchplate.github.io/openHASP-docs/0.6.2/integrations/home-assistant/howto/#configuration)
> 
> First prepare your plates to be integrated with Home Assistant (follow steps in order):
> 
>  - Connect your plates to the [network](https://haswitchplate.github.io/openHASP-docs/0.6.2/installation/wifi-setup/). Static DHCP or fixed IP is not needed **as communication only happens through MQTT.**
>   - Set the [GPIO configuration](https://haswitchplate.github.io/openHASP-docs/0.6.2/configuration/gpio/) corresponding to your hardware (important for them to be detected as entities), save and reboot.
>   - Restart Home Assistant.
>   - Set **the [MQTT server](https://haswitchplate.github.io/openHASP-docs/0.6.2/configuration/mqtt/) settings and make sure each plate has a unique node name, save and reboot.**

Everyone else was able to follow these instructions step by step from the first moment.

**Reply by filmchaser1** [reply]
Thank you for the reply. As a new user of this device, and again specific to wt32-01 I read the information in order. The first thing it says "INSTALL" and gives instructions that don't work in Home Assistant. The next set you quoted falls under "Configuration". One would assume you have to first "install" before "configuring", and so the endless loop...yes,"optional" is in the phrase, however, it doesn't distinguish and reads as the optional part is the method of installing through HACS or manual, but implies it must be installed as part of the setup first. I was frustrated and admitted that, now I'm trying to help anyone else running into the issue. A line a the top of that page saying "You must install an MQTT broker of choice ie:Mosquito within Home Assistant." Would have helped alot. I still don't know what HACS had to do with it at all but that's what hung  me up.

Installation[~](https://haswitchplate.github.io/openHASP-docs/0.6.2/integrations/home-assistant/howto/#installation)[](https://github.com/custom-components/hacs)
**You have the option to install the custom component using [HACS](https://hacs.xyz/) or via manual download:**


Using HACS
Manual Install
hacs_badge

Install using HACS in one-click. This is the preferred and recommended method, as HACS provides a very effective way to keep the component updated and/or choose between various versions.

Goto Home Assistant > HACS > Integrations.
Click the Explore & Add Repositories button.
Search for openHASP and click on the openHasp logo.
Click Install this repository in HACS. Note: To install the current unstable development version select the Main.
Click Install
Reboot Home-Assistant
Warning

You have to use component version consistently with the firmware version on your plates. For example, if your plates are at firmware version 0.6.x, you also need to use component version 0.6.x to ensure interoperability. Home Assistant will show a warning if it finds a version mismatch. Note that you can only have one version of the component installed at a time so a mix of plate versions cannot be deployed.

Configuration[~](https://haswitchplate.github.io/openHASP-docs/0.6.2/integrations/home-assistant/howto/#configuration)
First prepare your plates to be integrated with Home Assistant (follow steps in order):


Also, unless I'm missing something else, they device does not show up in "The component will automatically discover the plates and you will see them appearing in Lovelace UI's Configuration > Devices & Services > openHASP." The MQTT broker does but the plate nor "openHASP" does. I see communication on events so I know the plate is connected but it does not appear in configuration or Devices tabs.
![image](https://user-images.githubusercontent.com/99110891/152698438-ead0f8a9-aa90-4871-924e-6f24ee72587c.png)
![image](https://user-images.githubusercontent.com/99110891/152698619-2cae5eba-9a89-46d9-a412-13d04998dcf6.png)

**Reply by filmchaser1** [reply]
UPDATE: OK, so it seems that the instructions assume alot from the HA user. I found that I had to install an SSH client, to then install HACS so I would have the option to look in HACS to find the openHASP add-on to install so it can be discovered and populated in HA. I'm almost there, and again, hopefully this helps anyone else jumping into HA and openHASP from the beginning.

**Reply by fvanroie** [reply]
> I'm almost there, and again, hopefully this helps anyone else jumping into HA and openHASP from the beginning.

Thanks for your input @filmchaser1. The best way to improve the documentation is to use the ✏️-icon at the top of each page. It will open an editor on GitHub where you can apply the proposed changes directly in the openHASP-docs repository. 

This way the changes can be done where they need to be applied. I don't think users will search the WT32 SC01 discussion for help with the HACS installation.
